src/CMakeLists.txt

```diff
@@ -204,18 +204,18 @@ endif()
 
 if ( EXIV2_ENABLE_XMP )
     #target_include_directories(exiv2lib PRIVATE ${CMAKE_SOURCE_DIR}/xmpsdk/include)
-    target_link_libraries(exiv2lib PUBLIC xmp)
+    target_link_libraries(exiv2lib PRIVATE xmp)
```

Comment:
I think this is the culprit of the linking error detected by AppVeyor:

```
[154/168] Linking CXX executable bin\geotag.exe
FAILED: bin/geotag.exe 
cmd.exe /C "cd . && "C:\Program Files (x86)\CMake\bin\cmake.exe" -E vs_link_exe --intdir=samples\CMakeFiles\geotag.dir --manifests  -- C:\PROGRA~2\MI0E91~1.0\VC\bin\amd64\link.exe /nologo samples\CMakeFiles\geotag.dir\geotag.cpp.obj  /out:bin\geotag.exe /implib:lib\geotag.lib /pdb:bin\geotag.pdb /version:0.0  /machine:x64 /INCREMENTAL:NO  /subsystem:console -LIBPATH:C:\projects\exiv2-wutfp\build\install\lib lib\exiv2.lib C:\Users\appveyor\conanCache\zlib\1.2.11\lasote\stable\package\63da998e3642b50bee33f4449826b2d623661505\lib\zlibstatic.lib C:\.conan\fvzhjf\1\lib\libcurl_imp.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
LINK: command "C:\PROGRA~2\MI0E91~1.0\VC\bin\amd64\link.exe /nologo samples\CMakeFiles\geotag.dir\geotag.cpp.obj /out:bin\geotag.exe /implib:lib\geotag.lib /pdb:bin\geotag.pdb /version:0.0 /machine:x64 /INCREMENTAL:NO /subsystem:console -LIBPATH:C:\projects\exiv2-wutfp\build\install\lib lib\exiv2.lib C:\Users\appveyor\conanCache\zlib\1.2.11\lasote\stable\package\63da998e3642b50bee33f4449826b2d623661505\lib\zlibstatic.lib C:\.conan\fvzhjf\1\lib\libcurl_imp.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:bin\geotag.exe.manifest" failed (exit code 1120) with the following output:
geotag.cpp.obj : error LNK2019: unresolved external symbol XML_ParserCreate referenced in function "bool __cdecl readXML(char const *,class Options &)" (?readXML@@YA_NPEBDAEAVOptions@@@Z)
geotag.cpp.obj : error LNK2019: unresolved external symbol XML_SetElementHandler referenced in function "bool __cdecl readXML(char const *,class Options &)" (?readXML@@YA_NPEBDAEAVOptions@@@Z)
geotag.cpp.obj : error LNK2019: unresolved external symbol XML_SetCharacterDataHandler referenced in function "bool __cdecl readXML(char const *,class Options &)" (?readXML@@YA_NPEBDAEAVOptions@@@Z)
geotag.cpp.obj : error LNK2019: unresolved external symbol XML_SetUserData referenced in function "bool __cdecl readXML(char const *,class Options &)" (?readXML@@YA_NPEBDAEAVOptions@@@Z)
geotag.cpp.obj : error LNK2019: unresolved external symbol XML_Parse referenced in function "bool __cdecl readXML(char const *,class Options &)" (?readXML@@YA_NPEBDAEAVOptions@@@Z)
geotag.cpp.obj : error LNK2019: unresolved external symbol XML_ParserFree referenced in function "bool __cdecl readXML(char const *,class Options &)" (?readXML@@YA_NPEBDAEAVOptions@@@Z)
bin\geotag.exe : fatal error LNK1120: 6 unresolved externals
```
Some of the sample applicaitons are using XMP symbols directly and we were exposing them through the *exiv2lib* module. Maybe we can keep this private and link against xmp directly in the sample applications depending on it. Do you have a windows workstation where to try those changes? Otherwise I can try it by myself.